File: ui/src/modules/auth/events.ts
import { bus, store } from "@/core";
import * as shared from "@/shared";
import { actions } from "./store";
import { actions as loginActions } from "./login";

bus.on(shared.events.UI_AUTH_SHOW, payload => {
  store.dispatch(actions.setShow(payload));
  if (payload) store.dispatch(loginActions.setShow(payload));
});

File: client/csharp/Account/Login/Events.cs
using RAGE;
using Newtonsoft.Json;

namespace Project.Client.Account.Login
{
    public class Events : RAGE.Events.Script
    {
        public Events()
        {
            RAGE.Events.Add(Shared.Events.UI_LOGIN_SUBMIT_OK, OnUiLoginSubmitOk);
        }

        public static void OnUiLoginSubmitOk(object[] args)
        {
            var payload = JsonConvert.DeserializeObject<Shared.Schemes.UiLoginSubmitPayload>((string)args[0]);

            if (payload.Remember)
            {
                RAGE.Events.CallLocal("JS_ACCOUNT_STORE_CREDENTIALS", payload);
            }
            else
            {
                RAGE.Events.CallLocal("JS_ACCOUNT_REMOVE_CREDENTIALS");
            }

            Bus.TriggerUi(Shared.Events.UI_LOGIN_SUBMIT_OK);
        }
    }
}
